### PR TITLE
Advanced tensor ops for filtering heuristics

### PR DESCRIPTION
**Pull request by supersergiy**

### CODE REVIEW

File: zetta_utils/tensor_ops/mask.py [hunk 1]
```diff
@@ -287,3 +288,31 @@ def kornia_dilation(
...
+@builder.register("combine_mask_fns")  # type: ignore
+@supports_dict
+@skip_on_empty_data
+@typechecked
+def combine_mask_fns(
+    data: TensorTypeVar, fns: Sequence[Callable[[TensorTypeVar], TensorTypeVar]]
+) -> TensorTypeVar:  # pragma: no cover # no logic
+    assert len(fns) > 0
+    masks_t = [convert.to_torch(fn(data)) for fn in fns]
+    result_t = reduce(torch.Tensor.add_, masks_t, torch.zeros_like(masks_t[0]))
```
Comment (nkemnitz) [inline zetta_utils/tensor_ops/mask.py:316]: This should either be `Tensor.maximum_` or needs to handle potential overflow issues by widening the data type.

File: zetta_utils/tensor_ops/mask.py [hunk 2]
```diff
@@ -287,3 +288,31 @@ def kornia_dilation(
...
+    return result
+
+
+@builder.register("combine_mask_fns")  # type: ignore
+@supports_dict
+@skip_on_empty_data
+@typechecked
+def combine_mask_fns(
+    data: TensorTypeVar, fns: Sequence[Callable[[TensorTypeVar], TensorTypeVar]]
+) -> TensorTypeVar:  # pragma: no cover # no logic
```
Comment (nkemnitz) [inline zetta_utils/tensor_ops/mask.py:313]: Don't agree with the `no logic` here - see overflow issue below.

File: zetta_utils/tensor_ops/mask.py [hunk 3]
```diff
@@ -287,3 +288,31 @@ def kornia_dilation(
...
+    assert len(fns) > 0
```
Comment (nkemnitz) [inline zetta_utils/tensor_ops/mask.py:314]: Return `zeros_like(data)`, or raise `ValueError`?

**Review by nkemnitz — CHANGES_REQUESTED** [review]
